src/main/java/org/spongepowered/common/mixin/api/mcp/entity/monster/PatrollerEntityMixin_API.java

```diff
@@ -25,18 +25,37 @@
 package org.spongepowered.common.mixin.api.mcp.entity.monster;
 
 import net.minecraft.entity.monster.PatrollerEntity;
+import net.minecraft.util.math.BlockPos;
+import org.spongepowered.api.data.value.Value;
 import org.spongepowered.api.entity.living.monster.Patroller;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Shadow;
 
+import java.util.Set;
+
 @Mixin(PatrollerEntity.class)
 public abstract class PatrollerEntityMixin_API extends MonsterEntityMixin_API implements Patroller {
 
+    @Shadow private BlockPos patrolTarget;
     @Shadow public abstract void shadow$resetPatrolTarget();
 
     @Override
     public void findPatrolTarget() {
         this.shadow$resetPatrolTarget();
     }
 
+    @Override
+    protected Set<Value.Immutable<?>> api$getVanillaValues() {
+        final Set<Value.Immutable<?>> vanillaValues = super.api$getVanillaValues();
+
+        if (this.patrolTarget != null) {
```

Comment:
We'll need to add Optional<Value<Something>> to the Entity interfaces that have Values that may not exist so that everything goes through the DataProvider system.

Reply:
Remove the if and the method `targetPosition` is going through the provider system.